### ISSUE TITLE
Server: Authentication API

### ISSUE THREAD

**Issue by RARM** [RARM/Lynx#14]
Ensure the authentication API (login, sign up, and log out) works. Please, double-check the endpoint works (e.g., use curl to test all endpoints). Create some automated tests (this will help with any changes along the way in the coming days).

**Comment by RARM** [comment]
I added some documentation for the [authentication API](https://github.com/RARM/Lynx/wiki/Authentication-API) to make writing tests easier.

**Comment by RARM** [comment]
I wrote a module for handling user authentication and requests (so the server would recognize the same user) for the client. I created tests for this module. They should be enough for testing the endpoints and the new module.

Although it is two tests only, they are comprehensive tests.

Note that I didn't use Django tests in this case. I didn't do so because we are using a third-party module. I recommend creating a wrapper and testing for that wrapper. But that may be overkill for this soon-to-due and small project.

<img width="447" alt="Screenshot 2023-08-02 at 8 02 07 PM" src="https://github.com/RARM/Lynx/assets/30755087/a67bda54-a034-40bd-b1e6-c1fe41a249ec">
